pandas/core/arrays/categorical.py

```diff
@@ -2001,12 +2012,17 @@ def __repr__(self) -> str:
         String representation.
         """
         _maxlen = 10
+        na_repr = "NaN"
+        if is_extension_array_dtype(self.categories.dtype):
```

Comment:
heads up: we've moving away from is_foo_dtype, so this would be `isinstance(..., ExtensionDtype)`

Reply:
Thanks.